p2p/discovery.py

```diff
@@ -765,6 +791,45 @@ def process(received_echo: Hash32, response: List[kademlia.Node]) -> None:
 
         return tuple(n for n in nodes if n != self.this_node)
 
+    async def get_ticket(self, node: kademlia.Node, topics: List[bytes]) -> 'Ticket':
+        token = self.send_ping_v5(node, topics)
+        try:
+            got_pong, raw_pong, wait_periods = await self.wait_pong_v5(node, token)
+        except AlreadyWaitingDiscoveryResponse:
+            self.logger.debug("failed to get ticket from %s, already waiting for pong", node)
+            return None
```

Comment:
I'm generally :+1: on APIs that require return value checking as they are prone to errors when someone fails to actually check.

Reply:
I guess you mean :-1:?

Reply:
Anyway, I've changed it to raise an exception instead

Reply:
:joy_cat: yup, :-1: